client/reader/controller.js

```diff
@@ -148,6 +151,38 @@ module.exports = {
 		next();
 	},
 
+	checkForColdStart: function( context, next ) {
+		const FeedSubscriptionStore = require( 'lib/reader-feed-subscriptions' );
+		const user = getCurrentUser( context.store.getState() );
+
+		if ( ! user ) {
+			next();
+			return;
+		}
+
+		if ( ! user.is_new_reader ) {
+			next();
+			return;
+		}
+
+		function checkSubCount( tries ) {
+			if ( FeedSubscriptionStore.getCurrentPage() > 0 || FeedSubscriptionStore.isLastPage() ) {
+				// we have total subs now, make the decision
+				if ( FeedSubscriptionStore.getTotalSubscriptions() < config( 'reader_cold_start_graduation_threshold' ) ) {
+					defer( page.redirect.bind( page, '/read/start' ) );
+				} else {
+					defer( next );
+				}
+			} else if ( tries > -1 ) {
+				FeedSubscriptionStore.once( 'change', checkSubCount.bind( null, --tries ) );
+			} else {
+				defer( next );
+			}
+		}
+
+		checkSubCount( 3 );
```

Comment:
oh, that's not the number of subs, it's the number of change events to consume before giving up. :)